amd_openvx_extensions/amd_rpp/source/tensor/ColorJitter.cpp

```
@@ -0,0 +1,264 @@
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+THE SOFTWARE.
+*/
+
+#include "internal_publishKernels.h"
+
+struct ColorJitterLocalData {
+    vxRppHandle *handle;
+    vx_uint32 deviceType;
+    RppPtr_t pSrc;
+    RppPtr_t pDst;
+    vx_float32 *pBrightness;
+    vx_float32 *pContrast;
+    vx_float32 *pHue;
+    vx_float32 *pSaturation;
+    RpptDescPtr pSrcDesc;
+    RpptDescPtr pDstDesc;
+    RpptROI *pSrcRoi;
+    RpptRoiType roiType;
+    vxTensorLayout inputLayout;
+    vxTensorLayout outputLayout;
+    size_t inputTensorDims[RPP_MAX_TENSOR_DIMS];
+    size_t outputTensorDims[RPP_MAX_TENSOR_DIMS];
+};
+
+static vx_status VX_CALLBACK refreshColorJitter(vx_node node, const vx_reference *parameters, vx_uint32 num, ColorJitterLocalData *data) {
+    vx_status status = VX_SUCCESS;
+    STATUS_ERROR_CHECK(vxCopyArrayRange((vx_array)parameters[3], 0, data->inputTensorDims[0], sizeof(vx_float32), data->pBrightness, VX_READ_ONLY, VX_MEMORY_TYPE_HOST));
+    STATUS_ERROR_CHECK(vxCopyArrayRange((vx_array)parameters[4], 0, data->inputTensorDims[0], sizeof(vx_float32), data->pContrast, VX_READ_ONLY, VX_MEMORY_TYPE_HOST));
+    STATUS_ERROR_CHECK(vxCopyArrayRange((vx_array)parameters[5], 0, data->inputTensorDims[0], sizeof(vx_float32), data->pHue, VX_READ_ONLY, VX_MEMORY_TYPE_HOST));
+    STATUS_ERROR_CHECK(vxCopyArrayRange((vx_array)parameters[6], 0, data->inputTensorDims[0], sizeof(vx_float32), data->pSaturation, VX_READ_ONLY, VX_MEMORY_TYPE_HOST));
+
+    void *roi_tensor_ptr;
+    if (data->deviceType == AGO_TARGET_AFFINITY_GPU) {
+#if ENABLE_OPENCL || ENABLE_HIP
```

[Blocking] GPU support for ColorJitter looks inconsistent: `query_target_support` and `ColorJitter_Register` advertise GPU affinity (and enable GPU buffer access when HIP is enabled), but both `refreshColorJitter` and `initializeColorJitter` immediately return `VX_ERROR_NOT_IMPLEMENTED` when `deviceType == AGO_TARGET_AFFINITY_GPU`. On a GPU-affinity context this will cause node initialization/execution to fail rather than falling back to CPU. Consider either implementing the GPU path (similar to other tensor kernels) or restricting this kernel to CPU-only (e.g., via `query_target_support` and the wrapper so `deviceType` is never GPU). Also note that when both `ENABLE_OPENCL` and `ENABLE_HIP` are disabled, `roi_tensor_ptr` is never initialized in the GPU branch before it is cast to `RpptROI*`.